src/plaquette/visualizer/__init__.py

```diff
@@ -93,16 +93,18 @@ def _mk_df(
             raise NotImplementedError("multi-round syndromes are unsupported.")
         v_items = []
         i = 0
-        for vert_idx, vert_data in enumerate(self.code.tanner_graph.nodes_data):
+        for vert_idx, vert_data in enumerate(self.code.tanner_graph.nodes):
             assert vert_data is not None
-            if vert_data.coords is None:
-                raise ValueError(f"vertex {vert_idx} has no coordinates")
+            # TODO: finalize this part:
+            if vert_data.pos is None:
+                vert_data.pos = graph.Position()
+            #     raise ValueError(f"vertex {vert_idx} has no coordinates")
```

Comment:
This TODO was added in the visualizer PR I made as a quick workaround, I assume that with the coordinate generation procedure now it can be restored to the original validation check, right?
```suggestion
            if vert_data.pos is None:
                raise ValueError(f"vertex {vert_idx} has no coordinates")
```

Reply:
Yes it can be :)